CMakeLists.txt

```diff
@@ -25,6 +25,7 @@ option(USE_DPDK "Enable DPDK Framework" OFF)
 option(USE_HDF5 "Build HDF5 output stages" OFF)
 option(USE_OMP "Enable OpenMP" OFF)
 option(USE_OLD_ROCM "Build for ROCm versions 2.3 or older" OFF)
+option(NO_MEMLOCK "Do not lock buffer memory (useful when running in Docker)" OFF)
```

Comment:
Long term I'd prefer if this was a config file option instead of something we had to bake into the binary.  i.e. it could be an optional `memlock: <default:true>` parameter in the `bufferFactory`.  However I also cannot really see needing this outside of a docker container, so it might not be worth the work to restructure at this point.

Reply:
Yeah. I wondered about doing that, but as you say it would be quite a restructure.